### PR TITLE
k8sutil: move TPR update to k8sutil

### PR DESCRIPTION
**Pull request by xiang90**
For status reporting.

### CODE REVIEW

File: pkg/util/k8sutil/tpr_util.go [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+)
+
+// UpdateClusterTPRObject updates the given TPR object.
+// ResourceVersion of the object MUST be set or update will fail.
+func UpdateClusterTPRObject(k8s *unversioned.Client, host, ns string, e *spec.EtcdCluster) (*spec.EtcdCluster, error) {
+	b, err := json.Marshal(e)
+	if err != nil {
+		return nil, err
+	}
+	req, err := http.NewRequest("PUT",
```
Comment (hongchaodeng) [inline pkg/util/k8sutil/tpr_util.go]: "PUT" -> http.MethodPut

File: pkg/util/k8sutil/tpr_util.go [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+	"fmt"
+	"net/http"
+
+	"github.com/coreos/etcd-operator/pkg/spec"
+
+	"k8s.io/kubernetes/pkg/client/unversioned"
+)
+
+// UpdateClusterTPRObject updates the given TPR object.
+// ResourceVersion of the object MUST be set or update will fail.
```
Comment (hongchaodeng) [inline pkg/util/k8sutil/tpr_util.go]: What does it mean "or update will fail"?
What I am seeing here is that if rv not set then update will always succeed..

Comment (xiang90) [inline pkg/util/k8sutil/tpr_util.go]: It should not. Can you verify?

Comment (xiang90) [inline pkg/util/k8sutil/tpr_util.go]: I tried it a while ago. update will fail if no rv (or rv is not equal to current rv) is given

**Comment by hongchaodeng** [comment]
@xiang90 I will try the update thing.

This PR LGTM

**Comment by xiang90** [comment]
Test failed due to flaky test. Merging.

**Comment by hongchaodeng** [comment]
From the logs, it seems like cluster has been running as expected but a little delayed.
```
cluster-name=test-etcd-l39fk pkg=cluster 
time="2016-12-26T07:19:47Z" level=warning msg="all etcd pods are dead. Trying to recover from a previous backup" cluster-name=test-etcd-l39fk pkg=cluster 
time="2016-12-26T07:19:47Z" level=info msg="cluster created with seed member (test-etcd-l39fk-0000)" cluster-name=test-etcd-l39fk pkg=cluster 
time="2016-12-26T07:19:52Z" level=info msg="Start reconciling" cluster-name=test-etcd-l39fk pkg=cluster 
time="2016-12-26T07:19:52Z" level=info msg="running members: test-etcd-l39fk-0000" cluster-name=test-etcd-l39fk pkg=cluster 
time="2016-12-26T07:19:57Z" level=info msg="Finish reconciling" cluster-name=test-etcd-l39fk pkg=cluster 
time="2016-12-26T07:19:57Z" level=error msg="fail to reconcile: grpc: timed out when dialing" cluster-name=test-etcd-l39fk pkg=cluster 
time="2016-12-26T07:20:02Z" level=info msg="Start reconciling" cluster-name=test-etcd-l39fk pkg=cluster 
time="2016-12-26T07:20:02Z" level=info msg="running members: test-etcd-l39fk-0000" cluster-name=test-etcd-l39fk pkg=cluster 
time="2016-12-26T07:20:02Z" level=info msg="Expected membership: test-etcd-l39fk-0000" cluster-name=test-etcd-l39fk pkg=cluster 
time="2016-12-26T07:20:02Z" level=info msg="added member (test-etcd-l39fk-0001)" cluster-name=test-etcd-l39fk pkg=cluster 
time="2016-12-26T07:20:02Z" level=info msg="Finish reconciling" cluster-name=test-etcd-l39fk pkg=cluster 
time="2016-12-26T07:20:07Z" level=info msg="Start reconciling" cluster-name=test-etcd-l39fk pkg=cluster 
time="2016-12-26T07:20:07Z" level=info msg="running members: test-etcd-l39fk-0000,test-etcd-l39fk-0001" cluster-name=test-etcd-l39fk pkg=cluster 
time="2016-12-26T07:20:07Z" level=info msg="Expected membership: test-etcd-l39fk-0000,test-etcd-l39fk-0001" cluster-name=test-etcd-l39fk pkg=cluster 
time="2016-12-26T07:20:07Z" level=error msg="fail to add new member (test-etcd-l39fk-0002): rpc error: code = 2 desc = etcdserver: re-configuration failed due to not enough started members" cluster-name=test-etcd-l39fk pkg=cluster 
time="2016-12-26T07:20:07Z" level=info msg="Finish reconciling" cluster-name=test-etcd-l39fk pkg=cluster 
time="2016-12-26T07:20:07Z" level=error msg="fail to reconcile: rpc error: code = 2 desc = etcdserver: re-configuration failed due to not enough started members" cluster-name=test-etcd-l39fk pkg=cluster 
```

**Pull request merged**